android/keyple-plugin/android-nfc/src/main/kotlin/org/eclipse/keyple/plugin/android/nfc/AbstractAndroidNfcReader.kt

```diff
@@ -333,29 +362,9 @@ internal object AndroidNfcReaderImpl : AbstractObservableLocalReader(AndroidNfcR
         }
     }
 
-    override fun enableNFCReaderMode(activity: Activity) {
-        if (nfcAdapter == null) {
-            nfcAdapter = NfcAdapter.getDefaultAdapter(activity)
-        }
-
-        val flags = flags
-
-        val options = options
-
-        Timber.i("Enabling Read Write Mode with flags : $flags and options : $options")
-
-        // Reader mode for NFC reader allows to listen to NFC events without the Intent mechanism.
-        // It is active only when the activity thus the fragment is active.
-        nfcAdapter?.enableReaderMode(activity, this, flags, options)
-    }
-
-    override fun disableNFCReaderMode(activity: Activity) {
-        nfcAdapter?.disableReaderMode(activity)
-    }
-
     override fun stopWaitForCardRemoval() {
         Timber.d("stopWaitForCardRemoval")
-        isWatingForRemoval = false
+        isWaitingForRemoval = false
```

Comment:
The working isn't clear for me,
The transition between the internal observation states is supposed to be managed at the AbstractObservableLocalReader  level!